SingularityService/src/main/java/com/hubspot/singularity/resources/RequestResource.java

```diff
@@ -125,6 +123,18 @@ private void submitRequest(SingularityRequest request, Optional<SingularityReque
     requestHelper.updateRequest(request, oldRequest, requestState, historyType, JavaUtils.getUserEmail(user), skipHealthchecks, message);
   }
 
+  private void checkForAuthorization(SingularityRequest request, Optional<SingularityRequest> oldRequest, SingularityAuthorizationScope scope) {
```

Comment:
could we keep the code as it was and add something like a `SingularityValidator.checkForAuthorizedChanges(Optional<SingularityRequest> old, SingularityRequest new, SingularityUser user)` method that we call after the two `checkForAuthorization()` calls?